src/Silverback.Integration/Messaging/Validation/MessageValidator.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 Sergio Aquilini
+// This code is licensed under MIT license (see LICENSE file for details)
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Silverback.Messaging.Validation
+{
+    /// <summary>
+    ///     Validates the message and throws a <see cref="MessageValidationException" /> depending on the
+    ///     <see cref="MessageValidationMode" />.
+    /// </summary>
+    internal static class MessageValidator
+    {
+        /// <summary>
+        ///     Validates the message.
+        /// </summary>
+        /// <param name="message">The message.</param>
+        /// <param name="validationMode">The message validation mode.</param>
+        /// <returns>
+        ///     Returns a tuple indicating whether the message is valid <see cref="Tuple{boolean, String}.Item1"/> and the validation errors <see cref="ValueTuple{boolean, String}.Item2"/> />.
+        /// </returns>
+        /// <exception cref="MessageValidationException">
+        ///     The thrown exception when validation mode is
+        ///     <see cref="MessageValidationMode.ThrowException" />.
+        /// </exception>
+        public static (bool IsValid, string? ValidationErrors) CheckMessageIsValid(
+            object message,
+            MessageValidationMode validationMode)
+        {
+            var validationContext = new ValidationContext(message);
+            var results = new List<ValidationResult>();
+            var validMessage = Validator.TryValidateObject(message, validationContext, results, true);
+
+            if (validMessage)
+            {
+                return (true, null);
+            }
+
+            var validationResults = string.Join(
+                string.Empty,
+                results.Select(validationResult => $"{Environment.NewLine}- {validationResult.ErrorMessage}"));
```

Comment:
Like this the parameter value will begin with a new line. It would have been better to add the extra new line in the message template actually. 👀